### ISSUE TITLE
hashOfTreeWithVersion crashes when applying binary delta update

### ISSUE THREAD

**Issue by Omgitsu** [sparkle-project/Sparkle#605]
this is using 1.10.0
I keep running into this issue when applying binary deltas for a current project.  I'm not sure what I'm doing wrong when it comes to the deltas. Deltas are created using the BinaryDelta tool, and signed with DSA keys.  The deltas will apply correctly from the command line, but when the app tries to apply them I get a crash in hashOfTreeWithVersion.  

I'm assuming that I'm not using the correct version numbers, but I'm unclear on what the correct version numbers should be.

Here is what the crash is dumping:

```
Crashed Thread:        16  Dispatch queue: com.apple.root.default-qos

Exception Type:        EXC_BAD_ACCESS (SIGSEGV)
Exception Codes:       EXC_I386_GPFLT

Thread 16 Crashed:: Dispatch queue: com.apple.root.default-qos
0   libsystem_c.dylib             	0x00007fff86fc5319 fts_read$INODE64 + 498
1   org.andymatuschak.Sparkle     	0x000000010863f418 hashOfTreeWithVersion + 150 (SUBinaryDeltaCommon.m:158)
2   org.andymatuschak.Sparkle     	0x000000010863e38d applyBinaryDelta + 1186 (SUBinaryDeltaApply.m:107)
3   org.andymatuschak.Sparkle     	0x000000010863fae1 -[SUBinaryDeltaUnarchiver applyBinaryDelta] + 258 (SUBinaryDeltaUnarchiver.m:29)
4   libdispatch.dylib             	0x00007fff8686b323 _dispatch_call_block_and_release + 12
5   libdispatch.dylib             	0x00007fff86866c13 _dispatch_client_callout + 8
6   libdispatch.dylib             	0x00007fff8686988f _dispatch_root_queue_drain + 935
7   libdispatch.dylib             	0x00007fff86877fe4 _dispatch_worker_thread3 + 91
8   libsystem_pthread.dylib       	0x00007fff91123a9d _pthread_wqthread + 729
9   libsystem_pthread.dylib       	0x00007fff911213dd start_wqthread + 13

Thread 16 crashed with X86 Thread State (64-bit):
  rax: 0x0000000000000000  rbx: 0x000060000028a960  rcx: 0x0000000000000003  rdx: 0x0000000000000009
  rdi: 0x000060000028a960  rsi: 0x000000013a6b0afc  rbp: 0x000000013a6b0b30  rsp: 0x000000013a6b0b00
   r8: 0x0000000000000000   r9: 0x00000000e0000000  r10: 0x00000000000057b8  r11: 0x00007fff86fc5127
  r12: 0x600007fef3360743  r13: 0x0000000000000000  r14: 0x000060000028a960  r15: 0x00007fef336250c0
  rip: 0x00007fff86fc5319  rfl: 0x0000000000010246  cr2: 0x000000010863e000
  
Logical CPU:     0
Error Code:      0x00000000
Trap Number:     13
```
Here are the relevant parts of my appcast.xml
```
<enclosure url="http://bit.ly/1hDlsrZ"
          sparkle:version="1.0.70"
          sparkle:shortVersionString="1.0.70"
          length="236027176"
          type="application/octet-stream"
          sparkle:dsaSignature="MCwCFGTzCHAiYEGukfSUyzgwdNf4lCPhAhQC+7d13opmXsD+BYMH7nOnJVG93Q=="/>
<sparkle:deltas>
<enclosure url="https://s3.amazonaws.com/mockulus/deltas/1.0.1.0.60-1.0.1.0.70.delta"
              sparkle:version="1.0.70"
              sparkle:deltaFrom="1.0.60"
              length="76672"
              type="application/octet-stream"
              sparkle:dsaSignature="MC0CFQCWQ+BspZogpMg+6GWuvbOrtEFGHQIULxpV4QirKwJdP042eh+rEWv79Bw="/>
</sparkle:deltas>
```
Info.plist values for the version that needs updating:
```
<key>CFBundleShortVersionString</key>
<string>1.0</string>
<key>CFBundleVersion</key>
<string>1.0.60</string>
```
Info.plist values for the latest version:
```
<key>CFBundleShortVersionString</key>
<string>1.0</string>
<key>CFBundleVersion</key>
<string>1.0.70</string>
```

**Comment by zorgiepoo** [comment]
I don't see an issue with the info.plist or appcast. #536 fixed a crash regarding binary deltas. I think it's likely related; the fix is in 1.11 now. Probably we should document this..

**Comment by Omgitsu** [comment]
looks like it was related.
1.11 fixes this issue.
